packages/amplication-server/src/core/resource/resource.service.ts

```diff
@@ -381,6 +389,97 @@ export class ResourceService {
     return resource;
   }
 
+  async createPreviewService({
+    args,
+    user,
+    nonDefaultPluginsToInstall,
+    requireAuthenticationEntity,
+  }: CreatePreviewServiceArgs): Promise<Resource> {
+    const { serviceSettings, gitRepository, ...rest } = args.data;
+    const resource = await this.createResource(
+      {
+        data: {
+          ...rest,
+          resourceType: EnumResourceType.Service,
+        },
+      },
+      user,
+      gitRepository,
+      "create resource"
+    );
+
+    await this.prisma.resourceRole.create({
+      data: { ...USER_RESOURCE_ROLE, resourceId: resource.id },
+    });
+
+    if (requireAuthenticationEntity) {
+      await this.entityService.createDefaultEntities(resource.id, user);
+      serviceSettings.authEntityName = USER_ENTITY_NAME;
```

Comment:
Shouldn't this come from a parameter from the caller? (the name of the auth entity)?